### PR TITLE
You Can No Longer Place Force Fields On The Same Tile As Yourself

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
Adds a check to the force field projector that prevents you from placing fields on the same tile as yourself.

## Why It's Good For The Game
Some people have been using this recently in fights to become temporarily unclickable, which is kinda bad if you ask me.

## Changelog
:cl:
tweak: you can no longer place force fields on the same tile as yourself to become god for 30 seconds
/:cl:

### CODE REVIEW

**Comment by trexter555** [comment]
ided pr



its still a good pr though

File: code/game/objects/items/devices/forcefieldprojector.dm [hunk 1]
```diff
@@ -36,6 +37,9 @@
 		return
 	if(get_dist(T,src) > field_distance_limit)
 		return
+	if(get_dist(T,src) < field_distance_min)
```
Comment (Jared-Fogle) [inline code/game/objects/items/devices/forcefieldprojector.dm:40]: If you're only checking for being on the same tile, you should probably only need to check `if (get_turf(src) == T)`

**Comment by ExcessiveUseOfCobblestone** [comment]
No var ; just check if target turf is user turf OR store the distance value instead of calculate it twice fnr

File: code/game/objects/items/devices/forcefieldprojector.dm (hunk 1 above)
Comment (Paxilmaniac) [reply code/game/objects/items/devices/forcefieldprojector.dm]: yes

**Comment by ATH1909** [comment]
You could alternatively just make it check to see if any mobs are already on that tile.

**Comment by TheMidnightRose** [comment]
Just pointing out this might mess with the crusher vortex trophy

**Comment by Pugie** [comment]
This item is really bad: it's not even useful for engineering, it only costs metal and glass to make and is early on in the tech tree. Each projector is small sized & can produce 3 forcefield walls each that prevent movement & can be used to trap people or stop fights in one click. They also can be used through walls & anywhere in your character's view. Engineering holobarrier and atmos holofan already exist, pretty sure this was only ever added for validhunting.

**Comment by necromanceranne** [comment]
It's also probably better for security. You can trap people with it and shoot them with a disabler.

**Comment by Paxilmaniac** [comment]
Last I saw you can't shoot anything past them.

**Comment by Dennok** [comment]
Better make it show mob over field and make it like full tile (or 4 border) window, mob cant exit from it or use something from it.

**Comment by Tharcoonvagh** [comment]
How Can Mirrors Be Real If Our Eyes Aren't Real?

**Comment by Pugie** [comment]
> Last I saw you can't shoot anything past them.

nope lasers and disablers etc work.

**Comment by Dennok** [comment]
> > Last I saw you can't shoot anything past them.
> 
> nope lasers and disablers etc work.

Until you click forcefield, is i memorise you can shoot windows and force fields when click on it.

**Comment by Paxilmaniac** [comment]
not stale you stinky bot

**Comment by CoffeeDragon16** [comment]
Good PR, I definitely see people abuse that mechanic, hope it gets merged 👍

**Comment by optimumtact** [comment]
`if these are all true and you are auto-staled anyway, you need to actively ask maintainers if your PR will be merged.`

**Pull request merged**